From c7e04f7b24b2a9f08683ae4b23aa323e0b82b53a Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Wed, 28 Jun 2017 11:34:46 -0700
Subject: [PATCH 0389/1200] board: thunderx: enable board early init config
 item

use board_early_init_r to call pci_init to probe all pci
devices before env is accessed.

Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 board/cavium/thunderx/thunderx.c | 3 ++-
 include/configs/thunderx_93xx.h  | 4 +++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/board/cavium/thunderx/thunderx.c b/board/cavium/thunderx/thunderx.c
index 981c1fd6de..5416c50bc5 100644
--- a/board/cavium/thunderx/thunderx.c
+++ b/board/cavium/thunderx/thunderx.c
@@ -61,10 +61,11 @@ DECLARE_GLOBAL_DATA_PTR;
 char thunderx_prompt[CONFIG_THUNDERX_PROMPT_SIZE] = "ThunderX> ";
 
 #ifdef CONFIG_BOARD_EARLY_INIT_R
-extern void eth_common_init(void);
+extern void eth_common_init();
 int board_early_init_r(void)
 {
 	eth_common_init();
+	pci_init();
 	return 0;
 }
 #endif
diff --git a/include/configs/thunderx_93xx.h b/include/configs/thunderx_93xx.h
index 645de5e332..af5e1c4251 100644
--- a/include/configs/thunderx_93xx.h
+++ b/include/configs/thunderx_93xx.h
@@ -384,11 +384,13 @@
 /* Use uboot SPI APIs for env variables */
 #define CONFIG_SPI_ENV
 
+/* Use early board init for pci_init call */
+#define CONFIG_BOARD_EARLY_INIT_R
+
 #if defined(CONFIG_ATF)
 #define CONFIG_ENV_IS_IN_ATF
 #define CONFIG_SYS_ENV_ATF_NOR
 #elif defined(CONFIG_SPI_ENV)
-#define CONFIG_BOARD_EARLY_INIT_R
 #define CONFIG_ENV_IS_IN_SPI_FLASH
 #define CONFIG_ENV_SECT_SIZE		(64 * 1024)
 #define CONFIG_ENV_SPI_MAX_HZ		12500000
-- 
2.22.0

